### ISSUE TITLE
Getting warning for API version

### ISSUE THREAD

**Issue by mohanpindyala** [cloudfoundry/cf-java-client#508]
Getting following warning while running client version #2.0.0.BUILD-SNAPSHOT

```Client supports API version 2.54.0 and is connected to server with API version 2.43.0. Things may not work as expected.```

Is client not backward compatible

**Comment by twoseat** [comment]
Hi @mohanpindyala. We're currently still in development for version 2 (hence it's a `BUILD-SNAPSHOT`), which means that we're not trying to maintain backwards compatibility. Instead we're trying to keep pace with the CF API, which you may have noticed is updated quite frequently. Having said that, we don't expect to see any problems when running against recent versions.

Given that the CF version you're running against is ~8 months old, you may want to consider upgrading if that's possible.

**Comment by lastras** [comment]
@twoseat On a related note, I am trying to use this java client sdk to programmatically (say) create a service but I don't have a choice of CF API - I need to interface with an existing cf installation (api.ng.bluemix.net, IBM's bluemix). To verify the version,  when I authenticate using the cf command line tool I get

API endpoint:   https://api.ng.bluemix.net (API version: 2.44.0)  

The problem I am having is that I get no output when trying this:

```CloudFoundryClient cfc = SpringCloudFoundryClient.builder()
   .host("api.ng.bluemix.net")
   .username(“username")
   .password(“password")
   .build();
CloudFoundryOperations cfo = DefaultCloudFoundryOperations.builder()
   .cloudFoundryClient(cfc)
   .organization(“org")
   .space(“space")
   .build();

cfo.services().listInstances().map(s->s.getName()).subscribe(System.err::println);
```

I have verified that changing the host to something incorrect results in an error, but changing the username/password to incorrect values result in exactly the same silent (no printout) behavior. Thus I did a
```
System.out.println(cfo.SUPPORTED_CLI_VERSION);
```
and obtained 2.54.0 as the version.

From your discussion above it would seem that the library may not be able to support older CF installations but I still need to interface to such older installation.
